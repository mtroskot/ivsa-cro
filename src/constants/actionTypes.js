export const userActionTypes = Object.freeze({
  AUTH_USER: 'AUTH_USER',
  AUTH_USER_SUCCESS: 'AUTH_USER_SUCCESS',
  AUTH_USER_ERROR: 'AUTH_USER_ERROR',
  LOG_OUT: 'LOG_OUT'
});

export const uiActionTypes = Object.freeze({
  START_ACTION: 'START_ACTION',
  STOP_ACTION: 'STOP_ACTION',
  REFRESH_ACTION_START: 'REFRESH_ACTION_START',
  REFRESH_ACTION_STOP: 'REFRESH_ACTION_STOP',
  TOGGLE_POPUP_MESSAGE: 'TOGGLE_POPUP_MESSAGE'
});

export const localeActionTypes = Object.freeze({
  CHANGE_LOCALE: 'CHANGE_LOCALE',
  CHANGE_LOCALE_SUCCESS: 'CHANGE_LOCALE_SUCCESS'
});

export const newsActionTypes = Object.freeze({
  FETCH_NEWS: 'FETCH_NEWS',
  FETCH_NEWS_SUCCESS: 'FETCH_NEWS_SUCCESS',
  FETCH_NEWS_ERROR: 'FETCH_NEWS_ERROR',
  PUBLISH_NEWS: 'PUBLISH_NEWS',
  PUBLISH_NEWS_SUCCESS: 'PUBLISH_NEWS_SUCCESS',
  UPDATE_NEWS: 'UPDATE_NEWS',
  UPDATE_NEWS_SUCCESS: 'UPDATE_NEWS_SUCCESS',
  DELETE_NEWS: 'DELETE_NEWS',
  DELETE_NEWS_SUCCESS: 'DELETE_NEWS_SUCCESS'
});
