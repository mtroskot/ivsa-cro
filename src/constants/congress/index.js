/**
 * Congress data
 */
export const congressPlan = Object.freeze([
  ['09:00 – 17:00  Aerodrom pickup', '08:00 – 19:00  Registration', '19:00 - 20:30  Welcome Dinner'],
  [
    '07:00 – 08:00  Breakfast',
    '08:00 – 09:00  Opening ceremony',
    '09:00 - 09:30  Buddy hour',
    '09:30 – 13:00  GA 1',
    '13:00 – 14:00  Lunch',
    '14:00 – 16:00  Lectures',
    '14:00 - 15:30  Oie workshop',
    '16:00 – 19:00  GA 1',
    '19:00 – 20:00  Free time',
    '20:30 - Cultural evening'
  ],
  [
    '09:00 - 10:00  Breakfast - faculty',
    '10:00 – 12:00  Lectures - faculty',
    '10:45 - 11:10  Oie lecture - faculty',
    '11:30 - 12:30  Oie workshop - faculty',
    '12:00 – 13:00  Lunch - faculty',
    '13:00 – 17:00  GA 2 - faculty',
    '17:00 – 18:30  International Stands - faculty',
    '19:30 – 21:00  Dinner - dorms',
    '21:00 - 22:30  Silent acution - dorms',
    '22:30 - Pub crawl / Karaoke party - city centar'
  ],
  [
    '07:00  Breakfast and Departure to city of Krapina, Varaždin and Trakošćan castle ',
    '12:00 - 13:30  Lunch',
    '19:00  Arrival back to Zagreb',
    '19:00 – 20:30 Dinner - dorms',
    '21:00  Movies under the stars - dorms'
  ],
  [
    '07:00 – 08:00  Breakfast - faculty',
    '08:00 - 11:00  GA 3 - faculty',
    '11:00 – 12:00  Lectures + coffee break - faculty',
    '12:00 – 13:00  Lunch - faculty',
    '13:00 - 15:00  GA 3 - faculty',
    '15:00 – 17:00  Workshops - faculty',
    '17:00 - 18:00  Free time',
    '18:30 – 19:30  Dinner - dorms',
    '20:30 - 23:00  Live auction - faculty',
    '23:00 - Afterparty - faculty'
  ],
  [
    '07:00 – 08:00  Breakfast - faculty',
    '08:00 - 13:00  GA 4 - faculty',
    '13:00 – 14:00  Lunch - faculty',
    '14:00 – 16:00  Lectures - faculty',
    '16:00 - 19:00  Sports day - faculty',
    '20:00 - 21:00  Dinner - dorms',
    '21:30 - Nightout - Katran club'
  ],
  [
    '07:00  ANIMAL DAY: Breakfast and Departure to Risnjak, Đakovo, Karlovac - dorms lunch box',
    '12:00 - 13:30 Lunch',
    '19:00  Arrival back to Zagreb',
    '19:00 – 20:30  Dinner - dorms',
    '20:30  Optional: Zagreb by night- Rakia bar and Melin - city centar'
  ],
  [
    '07:00 – 08:00  Breakfast - faculty',
    '08:00 - 08:30  Buddy hour - faculty',
    '08:30 – 14:00  GA 5 - faculty',
    '14:00 – 15:00  Lunch - faculty',
    '15:00 – 18:00  ExCo lectures and workshop - faculty',
    '19:00 – 20:00  Dinner - dorms',
    '20:00 - Quiz night - dorms'
  ],
  [
    '07:00 Breakfast and OPTIONAL TRIPS - Departure to Plitvice, Wine road, Paintball or Zagreb Zoo - dorms lunch box',
    '13:00 - 14:00  Launch',
    '18:00 Arrival back to Zagreb',
    '20:00 Formal dinner + Afterparty'
  ],
  [
    '08:00 – 09:00  Breakfast - faculty',
    '09:00 – 13:00  GA 6 - faculty',
    '13:00 – 14:00  Lunch - faculty',
    '14:00 - 16:00  GA 6 - faculty',
    '16:00 – 18:00  Workshops - faculty',
    '19:00 – 20:00  Dinner - dorms',
    '21:00 - Last party - faculty'
  ],
  [
    '07:00 – 08:00 Breakfast - faculty',
    '08:00 – 13:00 GA 7 - faculty',
    '13:00 – 14:00  Lunch - faculty',
    '14:00 – 18:00 Scavanger hunt - city centar',
    '19:00 – 20:00 Dinner - dorms',
    '21:00   White t-shirt party - faculty'
  ],
  ['07:00 – 10:00 Breakfast - dorms', '10:00  Departures']
]);
