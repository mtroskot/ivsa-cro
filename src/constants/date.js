export const weekDays = Object.freeze(['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday']);
