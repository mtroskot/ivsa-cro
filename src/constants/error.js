export const DEFAULT_ERROR = 'Something went wrong';
