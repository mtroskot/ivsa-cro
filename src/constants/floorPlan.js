export const floorPlan = Object.freeze([
  ['Ground floor - Big lecture hall'],
  ['Ground floor - Library', 'Basement- Student rooms'],
  [
    'Ground floor - Department of Pathophysiology',
    'First floor - Department of Physiology and Radiobiology ',
    'Second floor - Department of Pharmacology and Toxicology'
  ],
  [
    'Ground floor - Department of Animal Breeding and Livestock Production',
    'First floor - Department of Microbiology and Infectious diseases '
  ],
  ['First and second floor - Department of Anatomy and Histology'],
  ['Ground floor - Department of Pathology', 'First floor - Department of Chemistry and Biochemistry'],
  ['Department of Forensic and Judicial Veterinary Medicine'],
  ['Reproduction and Obstetrics Clinic'],
  ['Department of Parasitology and Parasitic Diseases with Clinic'],
  ['Surgery, Orthopaedics and Ophtalmology Clinic'],
  ['Department of Radiology, Ultrasound Diagnostic and Physical Therapy']
]);
