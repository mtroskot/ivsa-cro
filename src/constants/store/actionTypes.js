export const userActions = Object.freeze({
  AUTH_USER: 'AUTH_USER',
  AUTH_USER_SUCCESS: 'AUTH_USER_SUCCESS',
  AUTH_USER_ERROR: 'AUTH_USER_ERROR',
  LOG_OUT: 'LOG_OUT'
});

export const uiActions = Object.freeze({
  UI_START_LOADING: 'UI_START_LOADING',
  UI_STOP_LOADING: 'UI_STOP_LOADING',
  TOGGLE_POPUP_MESSAGE: 'TOGGLE_POPUP_MESSAGE'
});

export const localeActions = Object.freeze({
  CHANGE_LOCALE: 'CHANGE_LOCALE',
  CHANGE_LOCALE_SUCCESS: 'CHANGE_LOCALE_SUCCESS'
});

export const newsActions = Object.freeze({
  PUBLISH_NEWS: 'PUBLISH_NEWS',
  PUBLISH_NEWS_SUCCESS: 'PUBLISH_NEWS_SUCCESS',
  UPDATE_NEWS: 'UPDATE_NEWS',
  UPDATE_NEWS_SUCCESS: 'UPDATE_NEWS_SUCCESS'
});

export const appActions = Object.freeze({
  AUTO_SIGN_IN: 'AUTO_SIGN_IN',
  APP_START: 'APP_START'
});
