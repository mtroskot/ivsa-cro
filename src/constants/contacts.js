import { dimensions } from 'src/styles';
const { rem } = dimensions;
import { icons } from 'src/constants/icons';

export const contactsBeforeFeeze = [
  { name: 'Magdalena', number: '+385957487243' },
  { name: 'Tanja', number: '+385915982255' },
  { name: 'Morana', number: '+385989907371' },
  { name: 'Petra', number: '+385997210766' },
  { name: 'Damjan', number: '+385917939819' },
  { name: 'Robert', number: '+385958771155' },
  { name: 'Mihajlo', number: '+385981887602' },
  { name: 'Sunčana', number: '+385996404535' },
  { name: 'Lana', number: '+385955126115' },
  { name: 'Anja', number: '+385959141933' },
  { name: 'Jurica', number: '+385958653210' },
  { name: 'Valentina', number: '+385997514198' },
  { name: 'Josip', number: '+385989230290' },
  { name: 'Renata', number: '+385918998681' },
  { name: 'Ivana', number: '+385981889474' },
  { name: 'Elizabeta', number: '+385951986369' }
];

const freezedObjectArray = contactsBeforeFeeze.map(o => Object.freeze(o));
export const contacts = Object.freeze(freezedObjectArray);

export const contactButtons = [
  {
    buttonId: 1,
    func: 'call',
    iconProps: { name: icons.CALL, color: '#3366ff', size: 40 * rem }
  },
  {
    buttonId: 2,
    func: 'message',
    iconProps: { name: icons.CHATBOXES, color: '#3366ff', size: 40 * rem }
  }
];
