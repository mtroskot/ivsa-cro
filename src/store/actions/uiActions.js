import { uiActionTypes } from 'src/constants/actionTypes';
import { ObjectUtils } from 'src/utils';

export const startAction = (name, params) => {
  if (!ObjectUtils.isObject(params)) {
    throw Error(`type of ${typeof params} is not an object`);
  }
  return {
    type: uiActionTypes.START_ACTION,
    payload: {
      action: {
        name,
        params
      }
    }
  };
};

export const stopAction = name => ({
  type: uiActionTypes.STOP_ACTION,
  payload: { name }
});

export const refreshActionStart = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_START,
  payload: { refreshAction }
});

export const refreshActionStop = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_STOP,
  payload: { refreshAction }
});

export const togglePopupMessage = (message, position) => ({
  type: uiActionTypes.TOGGLE_POPUP_MESSAGE,
  payload: {
    message,
    position
  }
});
