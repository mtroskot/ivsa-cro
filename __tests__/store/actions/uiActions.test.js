import configureStore from 'redux-mock-store';
import * as uiActions from 'src/store/actions/uiActions';
import { uiActionTypes } from 'src/constants/actionTypes';

jest.mock('src/utils', () => {
  return {
    ObjectUtils: {
      isObject: jest
        .fn()
        .mockReturnValue(false)
        .mockReturnValueOnce(true)
        .mockReturnValueOnce(true)
    }
  };
});

const middlewares = [];
const mockStore = configureStore(middlewares);
// Initialize mockstore with empty state
const initialState = {};
const store = mockStore(initialState);

describe('UI actions tests', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('dispatches startAction action 1', () => {
    const name = 'name';
    const params = { id: 'id' };
    const expectedActions = [
      {
        type: uiActionTypes.START_ACTION,
        payload: {
          action: {
            name,
            params
          }
        }
      }
    ];
    store.dispatch(uiActions.startAction(name, params));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches startAction action 2, params undefined', () => {
    const name = 'name';
    const expectedActions = [
      {
        type: uiActionTypes.START_ACTION,
        payload: { action: { name } }
      }
    ];
    store.dispatch(uiActions.startAction(name));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches startAction action 2a,params null, error expected', () => {
    const name = 'name';
    expect(() => {
      store.dispatch(uiActions.startAction(name, null));
    }).toThrow();
  });

  it('dispatches startAction action 3,params of type string, error expected', () => {
    const name = 'name';
    const params = 'params';
    expect(() => {
      store.dispatch(uiActions.startAction(name, params));
    }).toThrow();
  });

  it('dispatches startAction action 4,params null, error expected', () => {
    const name = 'name';
    const params = null;
    expect(() => {
      store.dispatch(uiActions.startAction(name, params));
    }).toThrow();
  });

  it('dispatches stopAction action', () => {
    const name = 'name';
    const expectedActions = [
      {
        type: uiActionTypes.STOP_ACTION,
        payload: { name }
      }
    ];
    store.dispatch(uiActions.stopAction(name));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches refreshActionStart action', () => {
    const refreshAction = 'refreshAction';
    const expectedActions = [
      {
        type: uiActionTypes.REFRESH_ACTION_START,
        payload: { refreshAction }
      }
    ];
    store.dispatch(uiActions.refreshActionStart(refreshAction));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches refreshActionStop action', () => {
    const refreshAction = 'narefreshActionme';
    const expectedActions = [
      {
        type: uiActionTypes.REFRESH_ACTION_STOP,
        payload: { refreshAction }
      }
    ];
    store.dispatch(uiActions.refreshActionStop(refreshAction));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches togglePopupMessage action', () => {
    const message = 'message';
    const position = 'position';
    const expectedActions = [
      {
        type: uiActionTypes.TOGGLE_POPUP_MESSAGE,
        payload: { message, position }
      }
    ];
    store.dispatch(uiActions.togglePopupMessage(message, position));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
});
