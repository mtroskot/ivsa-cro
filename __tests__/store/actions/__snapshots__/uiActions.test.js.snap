// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI actions tests dispatches refreshActionStart action 1`] = `
Array [
  Object {
    "payload": Object {
      "refreshAction": "refreshAction",
    },
    "type": "REFRESH_ACTION_START",
  },
]
`;

exports[`UI actions tests dispatches refreshActionStop action 1`] = `
Array [
  Object {
    "payload": Object {
      "refreshAction": "narefreshActionme",
    },
    "type": "REFRESH_ACTION_STOP",
  },
]
`;

exports[`UI actions tests dispatches startAction action 1 1`] = `
Array [
  Object {
    "payload": Object {
      "action": Object {
        "name": "name",
        "params": Object {
          "id": "id",
        },
      },
    },
    "type": "START_ACTION",
  },
]
`;

exports[`UI actions tests dispatches startAction action 2, params undefined 1`] = `
Array [
  Object {
    "payload": Object {
      "action": Object {
        "name": "name",
        "params": undefined,
      },
    },
    "type": "START_ACTION",
  },
]
`;

exports[`UI actions tests dispatches stopAction action 1`] = `
Array [
  Object {
    "payload": Object {
      "name": "name",
    },
    "type": "STOP_ACTION",
  },
]
`;

exports[`UI actions tests dispatches togglePopupMessage action 1`] = `
Array [
  Object {
    "payload": Object {
      "message": "message",
      "position": "position",
    },
    "type": "TOGGLE_POPUP_MESSAGE",
  },
]
`;
